doc/build-helpers/trivial-build-helpers.chapter.md

```diff
@@ -84,37 +175,265 @@ writeTextFile {
+# Usage of `writeText`
+```nix
+# Writes my-file to /nix/store/<store path>
 writeText "my-file"
   ''
   Contents of File
   '';
+```
+:::
+
+This example is a simpler way to spell:
+
+```nix
+writeTextFile {
+  name = "my-file";
+  text = ''
+    Contents of File
+  '';
+}
+```
+
+### `writeTextDir` {#trivial-builder-writeTextDir}
+
+Writes a text file within a subdirectory of the store.
+
+`writeTextDir` takes two arguments: `path` and `text`, each of which should be a string.
+
+`path`
+
+: the destination within the Nix store path under which to create the file.
+
+`text`
+
+: the contents of the file.
+
+The store path will be a directory. The Nix store identifier will be generated based on various elements of the path.
+
+::: {.example #ex-writeTextDir}
+# Usage of `writeTextDir`
+```nix
 # Writes contents of file to /nix/store/<store path>/share/my-file
 writeTextDir "share/my-file"
   ''
   Contents of File
   '';
+```
+:::
+
+The example is a simpler way to spell:
+
+```nix
+writeTextFile {
+  name = "my-file";
+  text = ''
+    Contents of File
+  '';
+  destination = "share/my-file";
+}
+```
+
+### `writeScript` {#trivial-builder-writeScript}
+
+Writes a script within the store.
+
+`writeScript` takes two arguments: `name` and `text`, each of which should be a string.
+
+`name`
+
+: the name used in the Nix store path.
+
+`text`
+
+: the contents of the file.
+
+The created file is marked as executable.
+
+The store path will include the the name, and it will be a file. Any path separators and shell-reserved elements in the name are escaped to create the store path identifier.
+
+Here is an example.
+
+::: {.example #ex-writeScript}
+# Usage of `writeScript`
+```nix
 # Writes my-file to /nix/store/<store path> and makes executable
 writeScript "my-file"
   ''
   Contents of File
   '';
-# Writes my-file to /nix/store/<store path>/bin/my-file and makes executable.
-writeScriptBin "my-file"
+```
+:::
+
+The example is a simpler way to spell:
+
+```nix
+writeTextFile {
+  name = "my-file";
+  text = ''
+    Contents of File
+  '';
+  executable = true;
+}
+```
+
+### `writeScriptBin` {#trivial-builder-writeScriptBin}
+
+Writes a script within a "bin" subirectory of a subdirectory of the store.
+
+`writeScriptBin` takes two arguments: `name` and `text`, each of which should be a string.
+
+`name`
+
+: the name used in the Nix store path and within the file generated under the store path.
+
+`text`
+
+: the contents of the file.
+
+The created file is marked as executable.
+
+The file's contents will be put into `/nix/store/<store path>/bin/<name>`.
+
+The store path will include the the name, and it will be a directory. Any path separators and shell-reserved elements in the name are escaped to create the store path identifier.
+
+::: {.example #ex-writeScriptBin}
+# Usage of `writeScriptBin`
+```nix
+writeScriptBin "my-script"
   ''
-  Contents of File
+  echo "hi"
+  '';
+```
+:::
+
+The example is a simpler way to spell:
```

Comment:
```suggestion
This example is equivalent to:
```

(Idea: avoiding colloquialisms, for those who might not have English as a culturally-familiar language.)

(Provided as a suggestion only in case you find it okay to apply as is.)